packages/blockchain/src/index.ts

```diff
@@ -1149,6 +1151,30 @@ export default class Blockchain implements BlockchainInterface {
     return this._genesisBlock
   }
 
+  checkAndTransitionHardForkByNumber(number: bigint, td?: BigIntLike): void {
+    this._common.setHardforkByBlockNumber(number, td)
+
+    switch (this._common.consensusAlgorithm()) {
+      case ConsensusAlgorithm.Casper:
+        if (!(this.consensus instanceof CasperConsensus)) {
+          this.consensus = new CasperConsensus({ blockchain: this })
+        }
+        break
+      case ConsensusAlgorithm.Clique:
+        if (!(this.consensus instanceof CliqueConsensus)) {
+          this.consensus = new CliqueConsensus({ blockchain: this })
+        }
+        break
+      case ConsensusAlgorithm.Ethash:
+        if (!(this.consensus instanceof EthashConsensus)) {
+          this.consensus = new EthashConsensus({ blockchain: this })
+        }
+        break
+      default:
+        throw new Error(`consensus algorithm ${this._common.consensusAlgorithm()} not supported`)
+    }
+  }
```

Comment:
Whew, that's a hard one: a whole new API method introduced in a rebase-fix-PR. 😋 😬

Are we confident enough that we want to expose? Or would we rather want to keep this private (or rather: protected) for now?

Reply:
yea it was tricky one to find! well will move to `protected` :+1: